tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -583,6 +618,27 @@ class ScopedConvolutionDescriptor {
       LOG(FATAL) << "could not set cudnn convolution descriptor: "
                  << ToString(status);
     }
+    // NOTE(benbarsdell): This only applies if tensor op math is enabled
+    //                      and algo selection is set to kDefaultAlgorithm.
+    this->set_use_tensor_op_math(true);
+  }
+
+  void set_use_tensor_op_math(bool use_tensor_op_math) {
+#if CUDNN_VERSION >= 7000
+    cudnnMathType_t math_type =
+        (use_tensor_op_math ?
+         CUDNN_TENSOR_OP_MATH :
+         CUDNN_DEFAULT_MATH);
+    if (TensorOpMath<true>::IsEnabled()) {
```

Comment:
Out of curiosity, is there a legitimate reason why people don't want to use tensor_op when it is available? Since we have autotuning, it will never get slower. 

So if the problem is functional, we might need to consider whether we should make the control op/subgraph level or globally?

Reply:
We expect tensor ops to be sufficient wherever FP16 inputs are numerically feasible. The environment variable just provides a sanity check to quickly rule out tensor ops as a culprit when issues are encountered (e.g., during some conversion from fp32 to fp16). In this context, a global option is sufficient. Once the numerics of fp16 training are well understood, the opt-out can likely be removed entirely.